content/docs/api-reference/dvcfs.md

```diff
@@ -0,0 +1,66 @@
+# DvcFileSystem
+
+DvcFileSystem provides a pythonic file interface (aka
+[fsspec](https://filesystem-spec.readthedocs.io/)) for a DVC repo and provides a
+read-only filesystem-like operations like `ls`, `du`, `glob`, `get`, `download`,
+etc.
+
+DvcFileSystem provides a single view of all the files/directories in your
+repository, be it Git-tracked or DVC-tracked, or untracked (in case of a local
+repository). DvcFileSystem is smart to reuse the files in your cache directory
+(if present) and can otherwise stream/fetch the files from your default remote.
+
+### Basic Usage
+
+```py
+>>> from dvc.fs.dvc import _DvcFileSystem as DvcFileSystem
```

Comment:
This is temporary. Eventually, this will look something like:
```suggestion
>>> from dvc.api import DvcFileSystem
```

Or, better:
```suggestion
>>> from dvc.fs import DvcFileSystem
```

Reply:
When do we expect this change?

Reply:
This week. It’s not difficult, we just need to decide.

I find `dvc.fs` namespace more natural, but it’s controversial compared to `dvc.api`.
Let me know if you have any opinions/preference.

Reply:
No strong preference as long as it's not `_DvcFileSystem`!

Reply:
Wouldn't this be capitalized `DVCFileSystem`?

Reply:
yep, probably should be ... for consistency with DVCLive

Reply:
In camel casing, the abbreviations are usually treated the same as lowercase words.

From [Wikipedia/Camel_case#Programming_and_coding](https://en.wikipedia.org/wiki/Camel_case#Programming_and_coding):
> Programming identifiers often need to contain [acronyms and initialisms](https://en.wikipedia.org/wiki/Acronyms_and_initialisms) that are already in uppercase, such as "old HTML file". By analogy with the title case rules, the natural camel case rendering would have the abbreviation all in uppercase, namely "oldHTMLFile". However, this approach is problematic when two acronyms occur together (e.g., "parse DBM XML" would become "parseDBMXML") or when the standard mandates lower camel case but the name begins with an abbreviation (e.g. "SQL server" would become "sQLServer"). For this reason, some programmers prefer to treat abbreviations as if they were lowercase words and write "oldHtmlFile", "parseDbmXml" or "sqlServer".

Also, [Google Javascript Style Guide](https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined):
> lowercase everything (including acronyms), then uppercase only the first character of:
> a) … each word, to yield upper camel case, or
> b) … each word except the first, to yield lower camel case

Reply:
Good points, @skshetry how do you feel about consistency though? (DVCLive). I bothers me a bit. Are there popular examples in Python world with all upper acronyms?

Reply:
MLFlow?

Reply:
[`socketserver.TCPServer`](https://docs.python.org/3/library/socketserver.html#socketserver.TCPServer) and [`http.HTTPServer`](https://docs.python.org/3/library/http.server.html#module-http.server) comes to mind from Python.

Reply:
Also urllib's HTTPResponse (Django uses `HttpResponse`). Also `json.JSONDecoder`. But `HTTP`/`JSON`/`TCP` are common acronyms.

Reply:
[PEP8](https://peps.python.org/pep-0008/#descriptive-naming-styles):
> Note: When using acronyms in CapWords, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError.

[Pocoo Styleguide: flask/click, etc.](https://flask.palletsprojects.com/en/1.1.x/styleguide/#naming-conventions):
> Class names: CamelCase, with acronyms kept uppercase (HTTPWriter and not HttpWriter)

Reply:
Afaik, all fsspec implementations (which is Python) are upper case when using acronyms: `SSHFileSystem`, `GCSFileSystem`,  `IPFSFileSystem`, `OSSFileSystem`, etc

Reply:
> Are there popular examples in Python world with all upper acronyms?

scikit-learn has tons of examples using acronyms with all caps: `PLSCanonical`, `KNNImputer`, `RBFSampler`, `SGDClassifier`, `RANSACRegressor`, etc.